packages/widget-v2/src/state/chains.ts

```diff
@@ -8,16 +8,39 @@ import {
   chains as chainsInitiaRegistry,
   assets as assetsInitiaRegistry,
 } from "@initia/initia-registry";
+import { chainRegistryChainToKeplr } from "@chain-registry/keplr";
+import { ChainInfo } from "@keplr-wallet/types";
 
 export const chains = [
   ...chainsChainRegistry,
   ...chainsInitiaRegistry,
 ] as Chain[];
+
 export const assets = [
   ...assetsChainRegistry,
   ...assetsInitiaRegistry,
 ] as AssetList[];
 
+
+export const chainInfos = chains.map((chain) => {
+  try {
+    return chainRegistryChainToKeplr(chain, assets)
+  } catch (_error) {
+    return false
+  }
+}).filter(Boolean) as ChainInfo[]
```

Comment:
I don't like this at all... I think we should refactor this

Reply:
also im a bit confused about the try catch `chainRegistryChainToKeplr` doesn't look like it returns a promise

Reply:
I'll test it myself and see

Reply:
@codingki let's refactor to this: we can use the getChainInfo function here and simply filter out undefineds
```ts
export const getChainInfo = (chainID: string) => {
  const chain = chains.find((chain) => chain.chain_id === chainID);
  if (!chain) return undefined;
  try {
    return chainRegistryChainToKeplr(chain, assets);
  } catch (_error) {
    return undefined;
  }
};

export const allChainInfo = chains
  .map((chain) => getChainInfo(chain.chain_id))
  .filter((info) => info);
```

Reply:
tbh `.filter(Boolean)` and `.filter((chainInfo) => chainInfo))` are exactly the same but for some reason i hate the first one it's unintuitive

Reply:
so `chainRegistryChainToKeplr` is not a promise but it can throw an error, so we need to wrap it with try catch. I can't find which chain is causing the error, but you can try converting `chains` from `chain-registry` it will show you the client side error. with try catch wrapper we can just not returning the chain if it has a convertion error

Reply:
yeah `undefined` is a good suggestion